### PR TITLE
docs: clarify environment variable and system service relationship for config

### PR DESCRIPTION
**Pull request by DavSanchez**
Please follow [conventional commit standards](https://www.conventionalcommits.org/en/v1.0.0/)
in your commit messages and pull request title.

## Give us some context

* What problems does this PR solve?
Clarifies how setting configuration options for the infrastructure-agent with environment variables work with system services.
* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.
https://newrelic.slack.com/archives/C5A2QGLKT/p1731946333684059

### CODE REVIEW

**Review by adutta-newrelic — CHANGES_REQUESTED** [review]

File: src/content/docs/infrastructure/infrastructure-agent/configuration/configure-infrastructure-agent.mdx [hunk 1]
```diff
@@ -89,6 +89,12 @@ Here are detailed descriptions of each configuration method:
     ```yml
     license_key: {{MY_LICENSE_KEY}}
     ```
+
+    <Callout variant="important">
+      Often the infrastructure agent will be running as a system service in the host, managed globally (e.g. `systemd` for Linux, Windows Services, `launchd` for macOS, etc).
+      To apply configuration changes with environment variables in this setup, these environment variables will need to be set and made visible to the isolated context where the infrastructure agent service is running.
+      The way of doing this is platform-dependent, check the documentation for your platform to add environment variables to system services.
```
[the remark this answers is not in the document]
Comment (DavSanchez) [reply src/content/docs/infrastructure/infrastructure-agent/configuration/configure-infrastructure-agent.mdx:96]: I'm mostly suggesting the documentation for the system manager of your platform (if you are on Linux then `systemd` docs, or Windows Services for Windows, etc). I'm adding a clarification.

File: src/content/docs/infrastructure/infrastructure-agent/configuration/configure-infrastructure-agent.mdx [hunk 2]
```diff
@@ -89,6 +89,12 @@ Here are detailed descriptions of each configuration method:
     ```yml
     license_key: {{MY_LICENSE_KEY}}
     ```
+
+    <Callout variant="important">
+      Often the infrastructure agent will be running as a system service in the host, managed globally (e.g. `systemd` for Linux, Windows Services, `launchd` for macOS, etc).
```
[the remark this answers is not in the document]
Comment (DavSanchez) [reply src/content/docs/infrastructure/infrastructure-agent/configuration/configure-infrastructure-agent.mdx]: Done!

File: src/content/docs/infrastructure/infrastructure-agent/configuration/configure-infrastructure-agent.mdx [hunk 3]
```diff
@@ -89,6 +89,12 @@ Here are detailed descriptions of each configuration method:
...
+      To apply configuration changes with environment variables in this setup, these environment variables will need to be set and made visible to the isolated context where the infrastructure agent service is running.
```
[the remark this answers is not in the document]
Comment (DavSanchez) [reply src/content/docs/infrastructure/infrastructure-agent/configuration/configure-infrastructure-agent.mdx]: Done!

File: src/content/docs/infrastructure/infrastructure-agent/configuration/infrastructure-agent-configuration-settings.mdx [hunk 4]
```diff
@@ -37,6 +37,16 @@ The [`license_key`](#license-key) is the only required setting.
 
 For an example of how all these variables can be used, see our [sample configuration template in GitHub](https://github.com/newrelic/infrastructure-agent/blob/master/assets/examples/infrastructure/newrelic-infra-template.yml.example).
 
+## Environment variables
+
+Many of the configuration options listed also admit using environment variables to set values, which will take precendence over the YAML configuration.
```
[the remark this answers is not in the document]
Comment (DavSanchez) [reply src/content/docs/infrastructure/infrastructure-agent/configuration/infrastructure-agent-configuration-settings.mdx]: Done!

File: src/content/docs/infrastructure/infrastructure-agent/configuration/infrastructure-agent-configuration-settings.mdx [hunk 5]
```diff
@@ -37,6 +37,16 @@ The [`license_key`](#license-key) is the only required setting.
...
 For an example of how all these variables can be used, see our [sample configuration template in GitHub](https://github.com/newrelic/infrastructure-agent/blob/master/assets/examples/infrastructure/newrelic-infra-template.yml.example).
 
+## Environment variables
+
+Many of the configuration options listed also admit using environment variables to set values, which will take precendence over the YAML configuration. 
+
+<Callout variant="important">
+  Often the infrastructure agent will be running as a system service in the host, managed globally (e.g. `systemd` for Linux, Windows Services, `launchd` for macOS, etc). To apply configuration changes with environment variables in this setup,
+  these environment variables will need to be set and made visible to the isolated context where the infrastructure agent service is running. The way of doing this is platform-dependent,
+  check the documentation for your platform to add environment variables to system services.
```
[the remark this answers is not in the document]
Comment (DavSanchez) [reply src/content/docs/infrastructure/infrastructure-agent/configuration/infrastructure-agent-configuration-settings.mdx]: Done!

**Review by adutta-newrelic — APPROVED** [review]
Changes approved.

**Pull request merged**